miden-proto/proto/store.proto

```diff
@@ -0,0 +1,102 @@
+// Specification of the database RPC.
+//
+// This provided access to the rollup data to the other nodes.
+syntax = "proto3";
+package store;
+
+import "account.proto";
+import "block.proto";
+import "digest.proto";
+import "note.proto";
+import "transaction.proto";
+
+// REQUESTS
+// ================================================================================================
+
+message GetBlockHeaderRequest {
+    block.BlockSpec block_spec = 1;
+}
+
+message GetNoteRequest {
+    note.NoteSpec note_spec = 2;
+}
+
+message GetTransactionDataRequest {
+    account.AccountId account_id = 1;
+    block.BlockSpec block_spec = 2;
+    repeated note.NoteOrigin notes = 3;
+}
+
+message ListNotesByTagRequest {
+    // Only include tags after this block.
+    block.BlockSpec block_spec = 1;
+    // Minimum value of the tag.
+    fixed64 tag_min = 2;
+    // Maximum value of the tag.
+    fixed64 tag_max = 3;
+}
+
+message SubmitTransactionRequest {
+    transaction.ProvenTransaction transaction = 1;
+}
+
+message CheckNullifiersRequest {
+    digest.Digest nullifier = 1;
+}
+
+// RESPONSE
+// ================================================================================================
+
+message GetBlockHeaderResponse {
+    block.BlockHeader header = 1;
+}
+
+message CheckNullifiersResponse {
+    bool result = 1;
+}
+
+message GetNoteResponse {
+    note.Note note = 1;
+}
+
+// Server response. The format of the data depends on the endpoint, it is the
+// responsability of the client to decode the data form its original format.
+message ResponseBytes {
+    bytes data = 1;
+}
+
+message ListNotesByTagResponse {
+    note.Note note = 1;
+}
+
+// DB API
+// ================================================================================================
+
+service Api {
+    // QUERY ENPOINTS
+    // --------------------------------------------------------------------------------------------
+
+    // Query a block header by block spec.
+    rpc GetBlockHeader(GetBlockHeaderRequest) returns (GetBlockHeaderResponse) {}
+
+    // Query a note by note spec.
+    rpc GetNote(GetNoteRequest) returns (GetNoteResponse) {}
+
+    // Query for nullifiers.
+    rpc CheckNullifiers(stream CheckNullifiersRequest) returns (stream CheckNullifiersResponse) {}
+
+    // Query data required for transaction execution.
+    rpc GetTransactionData(GetTransactionDataRequest) returns (ResponseBytes) {}
+
+    // STREAMING RESPONSE ENDPOINTS
+    // --------------------------------------------------------------------------------------------
+
+    // Lists notes matching the tag.
+    rpc ListNotesByTag(ListNotesByTagRequest) returns (stream ListNotesByTagResponse) {}
+
+    // DATA ENDPOINTS
+    // --------------------------------------------------------------------------------------------
+
+    // Submits a client side proven transaction.
+    rpc SubmitTransaction(SubmitTransactionRequest) returns (ResponseBytes) {}
```

Comment:
This endpoint would not be relevant for the store.

In general, `store` will have a subset of endpoints exposed via RPC (the ones used for data retrieval), but will also have additional endpoints needed for internal components (e.g., needed for transaction executor, block producer etc.).

Reply:
Do we have a workflow of how the transaction data is supposed to end at the store?

My rationale is that once the transaction is validated, it should be saved in the store too, for later retrieval

Reply:
removed